src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -3645,4 +3645,175 @@ public Long hstrlen(final String key, final String field) {
     return client.getIntegerReply();
   }
 
+  @Override
+  public EntryID xadd(final String key, final EntryID id, final Map<String, String> hash) {
+    return xadd(key, id, hash, Long.MAX_VALUE, true);
+  }
+  
+  @Override
+  public EntryID xadd(final String key, EntryID id, final Map<String, String> hash, final long maxLen, final boolean exactMaxLen) {
```

Comment:
In Redis, extra `~` means approximate length. So, IMHO, `approximateLength` is better as parameter than `exactMaxLen`.